### ISSUE TITLE
undetected missing algorithmx package when compiling komatufte

### ISSUE THREAD

**Issue by espinielli** [jolars/komadown#4]
This is to signal that the TeX Live mechanism to automagically install missing packages fails in the case of `algorithm`s feature as for the `komatufte` vignette.
Here is the error from the log file:

```
...
LaTeX Warning: Reference `alg:algorithm' on page 2 undefined on input line 527.



! LaTeX Error: Environment algorithm undefined.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.529 \begin{algorithm}
...
```

I am on a MS Windows machine and all missing fonts/styles got installed but `algorithmicx`.
I suppose this is due to the fact that [line 340](https://github.com/jolars/komadown/blob/master/inst/rmarkdown/templates/komatufte/resources/komatufte.tex#L340) in the `komatufte.tex` template (`\usepackage[noend]{algpseudocode}`)

```tex
% Algorithms
\IfFileExists{algpseudocode.sty}{%
\usepackage[noend]{algpseudocode}
...
```

will trigger TeX Live to search for `algpseudocode.sty` in CTAN and fail to discover it.
Once I manually installed `algorithmicx`, everything worked.

The following change makes the automatic installation and pdf production work magically again (tested by removing `algorithmicx` + failing PDF without the change + adding the change + success):

```tex
% Algorithms
\usepackage{algorithmicx}
\IfFileExists{algpseudocode.sty}{%
\usepackage[noend]{algpseudocode}
...
```

I am not sure this is a valid implementation...but it can make life easier for potential users...
HTH

**Comment by jolars** [comment]
The problem with this solution, however, is that the package will simply fail when algorithmicx is not available.

**Comment by espinielli** [comment]
Shouldn't it then be installed automatically?
I tested it with TinyTeX and all happens automatically...

**Comment by espinielli** [comment]
FYI I have been playing with another approach (search for scrbook inbookdown' not really needing an R package

**Comment by jolars** [comment]
In any case I think using `IfFileExists` is good practice and some people will not need the algorithm functionality. Have you tried putting \usepackage{algorithmicx} after IfFileExists?

**Comment by espinielli** [comment]
Yes I tried and it failed because `algpseudocode.sty` is part of `algorithmicx` package (see [CTAN](https://ctan.org/tex-archive/macros/latex/contrib/algorithmicx)).

When `algoritmicx` is not installed `algpseudocode.sty` is not found as a standalone .sty file by `tlmgr` when trying to (automagically) install it.
